wrapanapi/systems/rhevm.py

```diff
@@ -66,20 +68,37 @@ def _get_network(self, network_name):
         except IndexError:
             raise ItemNotFound('No match for network by "name={}"'.format(network_name))
 
-    def update_nic(self, network_name, nic_name='nic1', interface='VIRTIO'):
-        nic_service = self._get_nic_service(nic_name)
-        nic = nic_service.get()
+    def get_nics(self):
+        return self.api.nics_service().list()
+
+    def update_nic(self, network_name, nic_name='nic1', interface='VIRTIO', on_boot=True,
+                   add_nic=False, vnic_profile=None):
+        """Update or add a nic on VM/Template
+        Args:
+            network_name: string name of the network, also default for vnic_profile name if empty
+            nic_name: string name of the nic to add
+            interface: string interface type for pyVmomi, interfaces are resolved to a specific type
+            on_boot: boolean, kwarg for nic options
+            add_nic: boolean, if true will add a new nic instead of looking up existing one
+            vnic_profile: string name of the vnic_profile, nic_name is used if empty
+        """
+        # TODO take kwargs and match them to types.Nic attributes so callers can set any property
+        # generic nics service if adding, vm+nic specific service if updating
+        nic_service = self.api.nics_service() if add_nic else self._get_nic_service(nic_name)
+        # bare Nic type if adding, existing Nic object if updating
+        nic = types.Nic(name=nic_name) if add_nic else self._get_nic_service(nic_name).get()
         nic.network = self._get_network(network_name)
+        vnic_name = vnic_profile or network_name
```

Comment:
*REQUIRED*

Your docstring says `nic_name` is used if `vnic_profile` is empty but here you use `network_name`, so either the docstring or this line needs to change :)